[Nazryl] iP

Nazryl opened the pull request.

Nazryl committed:
Level-0

Nazryl committed:
Add features to Duke.java
* Echo commands
* Exit to exit program

Duke will now be called Allfather.
Update personality to Allfather.

Nazryl committed:
Update Duke.java
Commands are added into list. Display of items in list.

Update Allfather personality

Nazryl committed:
Update Duke.java
Add Task.java

Removal of array method to list method on adding task for compatibility with Task.java
Add checking of task and displaying of checking of task in list

Nazryl committed:
Add
*Able to add deadline, event and todo types of tasks
*Deadline.java
*Event.java
*TaskType.java
*Todo.java

Edit
*Added polymorphism technique to store task in a data structure
*Use inheritance to implement classes

Nazryl committed:
Enable Automated Text UI. Update tick and cross to Y and N in Task.Java

wangwaynesg commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,115 @@
+    }
+
+    public static void addTask(String commandType, String commandArgs) {
+        Task t = null;
```
Should Task t have a better variable name?

Nazryl replied:
Thank you, I shall implement a better name :)

wangwaynesg reviewed (commented):
Generally, I like the coding standards shown here and the use of static variables as commands. Very nice work!

Nazryl committed:
Catch invalid input commands for Todo, Deadline and Events
Refractor whole AllFather system
Fix bat file for result checking

Nazryl committed:
Refactor variables True False into CommandVariable

wangwaynesg commented on src/main/java/command/CommandVariable.java:
```diff
@@ -0,0 +1,38 @@
+	public static final String BOOLEAN_NO = "N";
+	public static final ArrayList<Task> taskList = new ArrayList<>();
+
+	public enum TaskType {
```
I like this! Interesting concept of enumerating TaskType here!

wangwaynesg commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,120 @@
+import exception.CommandException;
+import task.*;
+
+import java.util.ArrayList;
```
I like this! Use of ArrayList so that the list of tasks do not have a fixed capacity!

wangwaynesg reviewed (commented):
Overall very nice code! I learnt a few things reviewing this code and I might consider implementing them!

Nazryl committed:
Deletion of task is now available

Edit
*Usage of arraylist in project

Nazryl committed:
Loading of data is now available

Nazryl committed:
Deletion of task is now available

Edit
*Usage of arraylist in project

Nazryl committed:
Revert "Deletion of task is now available"

This reverts commit a275b9ef38718b80655bb8ca161b6dd4eebb5c67.

Nazryl committed:
File reading can catch errors

Nazryl committed:
Merge branch 'branch-Level-7'

Nazryl committed:
Merge branch 'master' of https://github.com/Nazryl/ip

Nazryl committed:
Saving and loading of Duke.txt into AllFather's ArrayList

leowweixiang commented on src/main/java/DataFile.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.Scanner;
+
+public class DataFile {
+	public static void loadData() {
```
This is TAB indentation, not 4-space. Please check your IntelliJ: File -> Settings -> Editor -> Code Style -> Java, make sure "Use tab character" is unchecked and "Tab size" is 4

Nazryl replied:
I will make the changes. May I know the particular reason for using a TAB indentation when it is already been set to default?

leowweixiang commented on src/main/java/DataFile.java:
```diff
@@ -0,0 +1,47 @@
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class DataFile {
```
Recommend a better name like DataFileManager, DataFileProcessor or something. DateFile suggests this is the data file, not a class to handle the data file

Nazryl replied:
Thank you for the suggestion. Changes have been made.

leowweixiang commented on src/main/java/DataFile.java:
```diff
@@ -0,0 +1,47 @@
+			}
+			CommandVariable.taskList.add(t);
+
+			if (splitArgs[1].equals("1")) {
```
Avoid magic literals, even if used only within this class; set up some static final constants

leowweixiang commented on src/main/java/command/CommandVariable.java:
```diff
@@ -0,0 +1,39 @@
+		D, E, T
+	}
+
+	public static void printHello() {
```
Will recommend putting such statements that are independent of command literals elsewhere, or renaming CommandVariable appropriately

leowweixiang commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,174 @@
+		Task task = CommandVariable.taskList.get(num-1);
+		task.markDone();
+
+		CommandVariable.printLine();
```
This hasn't been covered yet, but there is some frustration having to call some UI formatting methods from a class, then using System.out.println in the next step. So will recommend putting printing methods in one place

Nazryl replied:
I've made a print class.

leowweixiang commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,174 @@
+		System.out.println("Here are the tasks in your list:");
+
+		String filePath = "data/duke.txt";
+		File f = new File("data/duke.txt");
```
Seems that your write-to-file is only run when the user requests to print the list.

Can you try the following:
Run your program from a clean slate
Add a todo
Exit

Is the todo stored?

Nazryl replied:
I implement it such a way that when it is the user list, then the program will save the data. There were no specific instructions as to where and when is the save suppose to happen.

leowweixiang commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,174 @@
+			try {
+				commandInput = in.nextLine();
+				String commandType = commandInput.split(" ")[0];
+				String commandArgs = commandInput.substring(commandInput.indexOf(" ") + 1);
```
Recommend commandInput.split(" ",2)

Nazryl replied:
I have made the changes to store it under 1 array.

leowweixiang commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,174 @@
+
+	private static String generateString(Task t) {
+
+		int convertBool = t.getStatusIcon().equals("Y") ? 1 : 0;
```
Since this bool is about matching the status with "Y", I'll recommend using a more relevant name than convertBool

leowweixiang commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,174 @@
+			return CommandVariable.TaskType.E + " | " + convertBool + " | " + t.getDescription() + " | " + e.getScheduledDate();
+		default:
+		}
+		return null;
```
Will recommend returning "" instead of null, to avoid potential errant behaviour

leowweixiang commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,174 @@
+				break;
+			case CommandVariable.COMMAND_DEADLINE:
+				splitArgs = commandArgs.split(" /by ", 2);
+				if (splitArgs[0].toUpperCase().contains(CommandVariable.COMMAND_DEADLINE)) {
```
Recommend placing such validation in a separate method, or providing a throwing constructor that processes commandArgs

leowweixiang commented on src/main/java/command/CommandProcess.java:
```diff
@@ -0,0 +1,174 @@
+
+		CommandVariable.printHello();
+
+		loop: while (true) {
```
Recommend having a variable to terminate the loop rather than using a labeled break

Nazryl replied:
May I know why is this not recommended? It does the same job and having another variable is the same as having "loop" as a variable..

Nazryl committed:
Fix jar file reading and writing

Nazryl committed:
Update
*Refactoring of single use variables
*Save after every command
*Add error catching at file load, save, delete, mark task
*Refactor CommandException and DukeException class
*Multiple feedback from Github

Nazryl committed:
Update (More OOP)
*More refactoring and renaming of files to the appropriate folders
*Minimal classes are made to cater to specific task eg load/save, commandparser
*Task class to handle Arraylist commands

Nazryl committed:
Find tasks now available

Nazryl committed:
Event and deadline has a specific format to be input and a will convert to a new format for display

Nazryl committed:
Add header comments to addTask

Nazryl committed:
Merge pull request #1 from Nazryl/branch-Level-8

Merge Level-8

Nazryl committed:
Merge branch 'master' into branch-Level-9

Nazryl committed:
Merge pull request #2 from Nazryl/branch-Level-9

Merge Level-9

Nazryl committed:
Clean up merge codes

Nazryl committed:
Add more header comments to the rest of the project.

Nazryl committed:
Update README.md page

Nazryl committed:
Set theme jekyll-theme-modernist

Nazryl committed:
Add header comments to DukeException and CommandException

Nazryl committed:
Merge pull request #3 from Nazryl/branch-A-JavaDoc

Merge A-JavaDoc
